### DISCUSSION TITLE
Failed to visit Javadoc node

### DISCUSSION THREAD

**Discussion by Table-the-cat** [eclipse-jdt/eclipse.jdt.core#836] [Q&A] [answered]
I want to visit Javadoc node to get the SimpleName node in it, and I wrote code like this:
```
unit.accept(new ASTVisitor() {
            @Override
            public boolean visit(Javadoc doc) {
                doc.accept(new ASTVisitor() {
                    @Override
                    public boolean visit(SimpleName simpleName) {
                        types.add(simpleName.getIdentifier());

                        return true;
                    }
                });
                return true;
            }
        });
```
In which `unit` represented for a CompilationUnit, and `types` is a List that stores SimpleName nodes. Then I run the programme but can not get any SimpleName node in Javadoc.
I want to know why this code does not working and how to visit the Javadoc node in the AST?

**Comment by jjohnstn** [comment, marked as answer]
You need to use ASTVisitor(true) for the ASTVisitor constructor.  Otherwise, Javadoc-specific visit methods will return false by default and you won't get to the children.  They will return true by default if you specify true to the constructor so you don't need to override them.
